src/http_client_async.zig

```diff
@@ -2621,6 +2621,8 @@ pub fn onData(this: *HTTPClient, comptime is_ssl: bool, incoming_data: []const u
             if (this.state.content_encoding_i < response.headers.len and !this.state.did_set_content_encoding) {
                 // if it compressed with this header, it is no longer because we will decompress it
                 var mutable_headers = std.ArrayListUnmanaged(picohttp.Header){ .items = response.headers, .capacity = response.headers.len };
+                // we remove the content encoding header
+                _ = mutable_headers.orderedRemove(this.state.content_encoding_i);
```

Comment:
we can't remove the header here since it might be used by the user, see: https://github.com/oven-sh/bun/pull/5743

Reply:
If we keep the header we should not decode the body, because the user or libraries will see the header and try to decompress (like got) if the user wants the raw output (compressed) the user can use `decompress: false` options